packages/web-components/src/badge/badge.ts

```diff
@@ -0,0 +1,58 @@
+import { attr, FASTElement } from '@microsoft/fast-element';
+import { applyMixins, StartEnd } from '@microsoft/fast-foundation';
+import { BadgeAppearance, BadgeColor, BadgeShape, BadgeSize } from './badge.options.js';
+
+/**
+ * The base class used for constructing a fluent-badge custom element
+ * @public
+ */
+export class Badge extends FASTElement {
+  /**
+   * The appearance the badge should have.
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: appearance
+   */
+  @attr
+  public appearance: BadgeAppearance = BadgeAppearance.filled;
+
+  /**
+   * The color the badge should have.
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: color
+   */
+  @attr
+  public color: BadgeColor = BadgeColor.brand;
+  /**
+   * The shape the badge should have.
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: shape
+   */
+  @attr
+  public shape: BadgeShape = BadgeShape.circular;
+
+  /**
+   * The size the badge should have.
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: size
+   */
+  @attr
+  public size: BadgeSize = BadgeSize.medium;
+}
+
+/**
+ * Mark internal because exporting class and interface of the same name
+ * confuses API extractor.
+ * TODO: Below will be unnecessary when Badge class gets updated
+ * @internal
+ */
+/* eslint-disable-next-line */
+export interface Badge extends StartEnd {}
+applyMixins(Badge, StartEnd);
```

Comment:
I would appreciate more context and explanation for these.

Reply:
This will go away once the Badge base class is updated w/ Start/End slots. Ultimately start and end have `refs` in case someone needs to access the node or child nodes of start/end from a given class. So, there is a StartEnd class which enumerates the observables for these slots and here we are combining it with the Badge class to ensure that the final class includes everything from Badge + StartEnd support. The interface specifically exists to ensure typings are all good for "Badge" and I'm disabling that line because something in the lint config is converting it to a type.